### ISSUE TITLE
Document Visualization - Point Coordinates

### ISSUE THREAD

**Issue by ariskops** [MaartenGr/BERTopic#1746]
Hello,

Thank you for the wonderful work you have done and still do on BERTopic. It's a game-changer in topic analysis trully...

I am running a topic analysis for research purposes and I was wondering the following, is it possible to extract the "coordinates" of each document presented through the 'topic_visualize' function? The reason I am searching for this is because I would like to edit the plot on different data analysis software (e.g. Tableau). 

I already tried to extract the 'X, Y' of each point (per topic) by running the 'topic_model.visualize_documents(data_list).to_dict' command, but the final scatter plot is a tad different to the one generated by the script.  

Do you have any suggestions on this?

Thank you very much in advance!

**Comment by MaartenGr** [comment]
If you want to extract those coordinates you can simply extract the 2D-representations by using UMAP as shown in the code of the function itself. Something like this should suffice:

```python
reduced_embeddings = UMAP(n_neighbors=10, n_components=2, min_dist=0.0, metric='cosine').fit_transform(embeddings)
```

**Comment by ariskops** [comment]
Hello again,

Thank you for the prompt response. I tried your suggestion, but unfortunately this solution returns the plot of the UMAP dimensionality reduction. 

In my case, after running the UMAP (1) and HDBSCAN (2) models, I run the BERTopic model by setting umap_model=1 and hdbscan_model=2. At the end, the 'topic_model.visualize_documents(data_list)' command returns a plot showing each document on a D1-D2 axis system, coloured according to which topic BERTopic has assigned it. This plot is significantly different than the one deriving from the UMAP model. What I would like is to get the coordinates/embeddings of each point shown in the BERTopic visualisation and recreate it.

If you need any further details I am happy to share. I hope I was clear enough.

Thanks again for the support!

**Comment by MaartenGr** [comment]
The command `topic_model.visualize_documents(data_list)` essentially runs UMAP on the input embeddings as shown in the source code of `.visualize_documents`:

https://github.com/MaartenGr/BERTopic/blob/e2191f052d1461e95634ba52ccc44e4626a13db3/bertopic/plotting/_documents.py#L122-L123

Here, `embeddings_2d` are the X and Y points in 2-dimensional space that are plotted. Since in this case UMAP has no `random_state` state it will generate a different plot each time. You can prevent this by passing your own reduced UMAP embeddings to reproduce it.

**Comment by ariskops** [comment]
I see what you mean! I misunderstood something in the first read. Thank you for your patience. 

One final question; is it possible to use ParametricUMAP instead of UMAP as the dimensionality reduction function, in order to have a sense of the information lost during the 'compression' process? Would it make sense to you?

Thank you and I am sorry for any inconvenience.

**Comment by MaartenGr** [comment]
Sure, you can use whatever dimensionality reduction algorithm you want in BERTopic as long as it adheres to the API: https://maartengr.github.io/BERTopic/getting_started/dim_reduction/dim_reduction.html
